Remove duplicates from external links search results

richaagarwal opened the pull request:
This is a more simplified version of addressing what https://github.com/cfpb/cfgov-refresh/pull/4611 was attempting to address, by removing duplicates at the time of the search. 

See PR above and GHE/CFGOV/enhanced-cms/issues/39 for more context.

In addition to removing the duplicates:
  - We've updated the help text (thank you @marteki and @nataliafitzgerald)
  - Added tests (if you run those same tests off master currently, they will fail, as expected)

chosak commented on cfgov/search/views.py:
```diff
@@ -54,6 +56,14 @@ def post(self, request):
 
 
+def remove_duplicates(pages):
+        seen = set()
```
Can this please be made indented by 4 spaces to match consistency with the rest of the file?

marteki commented:
@richaagarwal - @nataliafitzgerald and I updated the helper text. I just pushed it up to this branch for you.

richaagarwal commented:
Thank you @marteki!  I have incorporated your changes and @chosak's feedback in my latest commits.

richaagarwal commented:
@chosak Thank you for the feedback!  I added a couple tests and changed the function to a static method.   

I also added an additional filter on the search since it seems unnecessary to search all pages -- just creatable ones -- in this commit: https://github.com/cfpb/cfgov-refresh/pull/4627/commits/479a7099f61561ccec479a35ac18f892795d3194.  Regardless of that, the duplicates are removed, but it seemed to make sense to narrow the scope of the search for performance.  

If you could take a look at the tests I added and that change I'd appreciate one final review!

chosak commented:
@richaagarwal 👍 tests look good! I have one minor suggestion that doesn't impact the testing logic at all but allows for a shorter and simpler test: instead of constructing the page and then using the  Wagtail StreamValue class to set the StreamField, set it as a constructor argument. See [this diff](https://github.com/cfpb/cfgov-refresh/compare/remove-dupes...chosak:remove-dupes-streamfield-content#diff-d6d072ad7619c97c3ffa408c81da4163L62) I made on a branch to demonstrate. To my eye this is a little more readable and straightforward about what the content of the StreamField is, and it also avoids using the internal StreamValue creation approach. I know we do it how you've done it here elsewhere in the codebase, though.

Regarding filtering out pages that aren't createable: do you have a sense for the performance benefit? The concern I'd have (especially if this goes undocumented) is that we might encounter a situation in future where we purposefully set `is_creatable` to false on a page type for the purposes of preventing its creation in the UI. Say we have a singleton page type (like the recent TDP index page) and we only ever want one of these ever, anywhere. So we might use a script to prevent it, and then use `is_creatable` to hide it from the UI. If we did that, those pages would disappear from the search results (or never appear if we did that from the beginning) and it wouldn't be clear to the user why that was.

richaagarwal commented:
@chosak Thanks! I've incorporated your updates to the tests; I agree it's more readable that way.  

And good point on `is_creatable`, I've removed that logic for now and can always revisit later if there's performance reasons to do so.
